addons/csi-vmware-cloud-director/driver.yaml

```diff
@@ -0,0 +1,456 @@
+  kind: ClusterRole
+  name: csi-provisioner-role
+subjects:
+  - kind: ServiceAccount
+    name: csi-vcd-controller-sa
+    namespace: kube-system
+---
+apiVersion: v1
+data:
+  vcloud-csi-config.yaml: |
+    vcd:
+      host: {{ required "Please provide VCD_URL" (trimSuffix "/api" .Credentials.VCD_URL) }}
+      org: {{ required "Please provide VCD_ORG" .Credentials.VCD_ORG }}
+      vdc: {{ required "Please provide VCD_VDC" .Credentials.VCD_VDC }}
+      vAppName: {{ default .Config.CloudProvider.VMwareCloudDirector.VApp .Params.vAppName }}
+    clusterid: {{ default .Config.Name .Params.clusterid }}
+kind: ConfigMap
+metadata:
+  name: vcloud-csi-configmap
+  namespace: kube-system
+---
+apiVersion: v1
+data:
+  password: '{{ with .Credentials.VCD_PASSWORD }}{{ . | b64enc }}{{ end }}'
+  refreshToken: '{{ with .Credentials.VCD_API_TOKEN }}{{ . | b64enc }}{{ end }}'
+  username: '{{ with .Credentials.VCD_USER }}{{ . | b64enc }}{{ end }}'
+kind: Secret
+metadata:
+  name: vcloud-basic-auth
+  namespace: kube-system
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: csi-vcd-controllerplugin
+  namespace: kube-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: csi-vcd-controllerplugin
+  template:
+    metadata:
+      annotations:
+        kubeone.k8c.io/credentials-hash: '{{ .CredentialsCCMHash }}'
+      labels:
+        app: csi-vcd-controllerplugin
+    spec:
+      affinity:
+        nodeAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - preference:
+                matchExpressions:
+                  - key: node-role.kubernetes.io/control-plane
+                    operator: Exists
+              weight: 1
+      containers:
+        - args:
+            - --csi-address=$(ADDRESS)
+            - --timeout=180s
+            - --v=5
+          env:
+            - name: ADDRESS
+              value: unix:///var/lib/csi/sockets/pluginproxy/csi.sock
+          image: '{{ .InternalImages.Get "VMwareCloudDirectorCSIAttacher" }}'
+          imagePullPolicy: IfNotPresent
+          name: csi-attacher
+          volumeMounts:
+            - mountPath: /var/lib/csi/sockets/pluginproxy/
+              name: socket-dir
+        - args:
+            - --csi-address=$(ADDRESS)
+            - --default-fstype=ext4
+            - --timeout=300s
+            - --v=5
+          env:
+            - name: ADDRESS
+              value: unix:///var/lib/csi/sockets/pluginproxy/csi.sock
+          image: '{{ .InternalImages.Get "VMwareCloudDirectorCSIProvisioner" }}'
+          imagePullPolicy: IfNotPresent
+          name: csi-provisioner
+          volumeMounts:
+            - mountPath: /var/lib/csi/sockets/pluginproxy/
+              name: socket-dir
+        - command:
+            - /opt/vcloud/bin/cloud-director-named-disk-csi-driver
+            - --cloud-config=/etc/kubernetes/vcloud/vcloud-csi-config.yaml
+            - --endpoint=$(CSI_ENDPOINT)
+            - --upgrade-rde
```

Comment:
Please remove this flag; it causes issues because this was designed specifically for the CSE(Kubernetes) offering from VMware Cloud Director. Which is something we don't use or rely upon.

Reply:
Previously it used to work fine but with the latest CSI driver this would cause the controller to fail/crash.

Reply:
done :+1:

Reply:
Damn it, now we can't even mark conversations as resolved? cc @xmudrii 😆

Reply:
I can :D